enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/HealthCheckEnricher.java

```diff
@@ -0,0 +1,109 @@
+package io.fabric8.maven.enricher.standard;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.DaemonSet;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.PetSet;
+import io.fabric8.maven.core.config.ProbeConfig;
+import io.fabric8.maven.core.handler.ProbeHandler;
+import io.fabric8.maven.enricher.api.AbstractHealthCheckEnricher;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by matthew.costa on 26/10/2016.
+ */
+public class HealthCheckEnricher extends BaseEnricher {
+
+    private ProbeConfig liveness;
+    private ProbeConfig readiness;
+    private ProbeHandler probeHandler;
+
+    public HealthCheckEnricher(EnricherContext buildContext) {
+        super(buildContext, "fmp-healthcheck");
+
+        probeHandler = new ProbeHandler();
+        liveness = buildContext.getResourceConfig().getLiveness();
+        readiness = buildContext.getResourceConfig().getReadiness();
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            List<Container> containers = null;
+
+            if(item instanceof Deployment) {
```

Comment:
Question is whether we want to allow that for ReplicationController and ReplicaSet, too ? (and how to work with OpenShift DeploymentConfigs, when given here). I guess we should add the health-checks here, too.